code/modules/modular_computers/file_system/computer_file.dm

```diff
@@ -31,9 +32,21 @@ var/global/file_uid = 0
 	temp.unsendable = unsendable
 	temp.undeletable = undeletable
 	temp.size = size
+	temp.password = password
 	if(rename)
 		temp.filename = filename + "(Copy)"
 	else
 		temp.filename = filename
 	temp.filetype = filetype
 	return temp
+
+/datum/computer_file/proc/can_access_file(var/mob/user, input_password = "")
+	if(!password)
+		return TRUE
+	else
+		if (!input_password)
+			input_password = sanitize(input(user, "Please enter a password to access file '[filename]':"))
```

Comment:
I think `file '[filename]'` is a bit redundant. What are you going to be accessing that isn't a file?

Reply:
Imho its fine.

> "Please enter a password to access"
Is what would be left and thats worse.

Maybe remove just the "file"
So its only:

> Please enter a password to file '[filename]':

Reply:
You're misunderstanding what I'm suggesting. Just removing the word 'file' is fine, so that it's `Please enter a password to access 'foo':`, or similar.